File: strategy/marvin/agent/base/simpleagent.lua
local Base = require "agent/base/agent"
local SimpleAgent = Class("Agent.Base.SimpleAgent", Base)

local debug = require "../base/debug"
local World = require "../base/world"


-- Child class must set _behaviors
-- SimpleAgent._behaviors = {}

function SimpleAgent:init(robot, messaging)
	Base.init(self, robot, messaging)
	self.beOffensive = false
end

function SimpleAgent.takeRobot(robots)
	for _, robot in ipairs(robots) do
		if SimpleAgent.checkRobot(robot) then
			return robot
		end
	end
end

function SimpleAgent.checkRobot(robot)
	return robot.isVisible and robot ~= World.FriendlyKeeper and not robot.userControl
end

function SimpleAgent:keepRobot()
	return self.checkRobot(self._robot)
end

function SimpleAgent:rateRobot()
	return 1
end

return SimpleAgent

File: strategy/typescript/glados/agent/base/simpleagent.ts
let Base from "glados/agent/base/agent"
let SimpleAgent = Class("Agent.Base.SimpleAgent", Base)

import * as World from "base/world";


// Child class must set _behaviors
// SimpleAgent._behaviors = {}

function SimpleAgent:init (robot, messaging) {
	Base.init(self, robot, messaging)
	this.beOffensive = false
}

function SimpleAgent.takeRobot (robots) {
	for (let robot of robots) {
		if (SimpleAgent.checkRobot(robot)) {
			return robot
		}
	}
}

function SimpleAgent.checkRobot (robot) {
	return robot.isVisible && robot != World.FriendlyKeeper && not robot.userControl
}

function SimpleAgent:keepRobot () {
	return this.checkRobot(this._robot)
}

function SimpleAgent:rateRobot () {
	return 1
}

return SimpleAgent
